lib/uuid_map.rb

```diff
@@ -34,4 +34,18 @@ def rewrite(data)
   def source
     [@newfile, @oldfile].find(&:exist?)
   end
+
+  def remap(from, to)
+    @data ||= mapping
+    check_ids(@data, from, to)
+    @data[to] = @data.delete(from)
+    rewrite(@data)
+  end
+
+  private
+
+  def check_ids(data, from, to)
+    abort "No existing data for #{from}" unless data[from]
+    abort "Already have data for #{to}" if data[to]
+  end
```

Comment:
This is a slightly odd helper function on several fronts:
* It has very different behaviour from what I'd expect a method called `check_ids` to do
* it's divorced from the actual object
* it's very tightly coupled to one particular usage pattern (ie it does separate things that aren't necessarily inter-dependent).
* `abort`-ing out of it is very aggressive (and as you've seen, causes problems with testing)

I think this would be a lot better as an actual method (i.e. that doesn't take `data` as an argument), and probably actually _two_ quite simple query methods: one for checking if a given `id` exists in the file, and one for checking if a given `uuid` exists in the file.